### DISCUSSION TITLE
Support a distinct_on filter

### DISCUSSION THREAD

**Discussion by zliebersbach** [ChilliCream/hotchocolate#4848] [Feature Request]
**Is your feature request related to a problem? Please describe.**
The problem is we there is no way to fetch data on distinct keys or nested entity properties.

**Describe the solution you'd like**
I'd like an implementation of `distinct_on` which allows for nested distinct queries. [See `Queryable.Distinct` documentation](https://docs.microsoft.com/en-us/dotnet/api/system.linq.queryable.distinct?view=netcore-3.1).

**Describe alternatives you've considered**
There are no alternatives. I do know that [Hasura supports this filter](https://hasura.io/docs/1.0/graphql/manual/queries/distinct-queries.html).

**Comment by zliebersbach** [comment]
P.S. Very happy to do offer development hours to help get this done, but I will need some direction. Through my initial poking around, I feel like it may make the most sense to either add the functionality to the `[UseFiltering]` attribute or create a new attribute, e.g. `[UseDistinct]`

**Comment by PascalSenn** [comment]
Hi @kkliebersbach 
I think this is very doable 👍  I guess it is not gonna be too much work. 

To investigate such a feature we usually do a comparison with already existing libraries.  Prisma, Hasura, GraphCMS, Craft, Gatsby etc.

Something like this:
https://github.com/ChilliCream/hotchocolate/issues/923#issuecomment-514002900

We then decide on which direction we take, write down the SDL and implement the feature. 

So its 
1. analysing existing solutions
2. Discussion and champion
3. Create SDL for the winner
4. Implement the feature
5. 🥳

**Reply by jamierytlewski** [reply]
Did this ever get implemented? I can't see it anywhere. Thanks for a great library.

**Comment by ArthNRick** [comment]
hello!
any update about this thread?

**Comment by Bowenf3** [comment]
bump

**Comment by ekarcnevets** [comment]
We are using and loving Hot Chocolate in our org, and having the ability to distinct results directly in the query would save us a lot of dev time adding additional resolvers for distinct queries or otherwise working around it.

Has this discussion progressed internally at all? If there is a clear way forward identified but no resource to tackle it, we could try and give implementing it a go?
